### PR TITLE
feat: where clauses recognize nested properties

### PR DESCRIPTION
**Pull request by pauldelucia**
## Issue being fixed or feature implemented

When building queries with WHERE clauses, by using document type properties field rather than flattened_properties, we weren't accounting for nested properties

## What was done?

Use flattened_properties instead of properties

## How Has This Been Tested?

Platform TUI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by ogabrielides — DISMISSED** [review]
LGTM

File: packages/rs-drive/src/query/conditions.rs [hunk 1]
```diff
@@ -1048,6 +1048,41 @@ impl<'a> WhereClause {
         Ok(query)
     }
 
+    /// Recursively find the property type for a given nested field name.
+    fn find_property_type(
+        document_type: &'a DocumentType,
+        field_name: &str,
```
Comment (lklimek) [inline packages/rs-drive/src/query/conditions.rs:1054]: function is about `property`, but this arg is named `field_name`. What's the difference between property and field?

File: packages/rs-drive/src/query/conditions.rs [hunk 2]
```diff
@@ -1048,6 +1048,41 @@ impl<'a> WhereClause {
...
+    ) -> Result<Cow<'a, DocumentPropertyType>, Error> {
+        let parts = field_name.split('.');
+
+        let mut current_properties = document_type.properties();
+        let mut current_property_type = None;
+
+        for part in parts {
```
Comment (lklimek) [inline packages/rs-drive/src/query/conditions.rs:1061]: looks a bit as use case for `Iterator::fold`

File: packages/rs-drive/src/query/conditions.rs [hunk 3]
```diff
@@ -1048,6 +1048,41 @@ impl<'a> WhereClause {
         Ok(query)
     }
 
+    /// Recursively find the property type for a given nested field name.
```
Comment (lklimek) [inline packages/rs-drive/src/query/conditions.rs:1051]: Please add / ask some AI for  proper documentation.

File: packages/rs-drive/src/query/conditions.rs (hunk 1 above)
Comment (pauldelucia) [reply packages/rs-drive/src/query/conditions.rs:1054]: Changed to property_name

File: packages/rs-drive/src/query/conditions.rs (hunk 2 above)
Comment (pauldelucia) [reply packages/rs-drive/src/query/conditions.rs]: I changed to use fold. I think it's harder to understand now. But it works the same. Let me know what you think.

Comment (pauldelucia) [reply packages/rs-drive/src/query/conditions.rs:1102]: Reverted

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**